src/Native/Unix/System.Native/pal_io.h

```diff
@@ -32,6 +33,24 @@ struct FileStatus
     int64_t Ino;       // inode number of the file
 };
 
+/**
+ * File status returned by Stat or FStat.
+ */
+struct FileStatus2
+{
+    int32_t Flags;     // flags for testing if some members are present (see FileStatusFlags)
+    int32_t Mode;      // file mode (see S_I* constants above for bit values)
+    uint32_t Uid;      // user ID of owner
+    uint32_t Gid;      // group ID of owner
+    int64_t Size;      // total size, in bytes
+    struct timespec ATime;     // time of last access
```

Comment:
Is the timespec guaranteed to be two 64-bit numbers? Should be rather define our own that we control the layout of?

Reply:
Posix defines the parts as time_t (which we already assume is int64_t) and long. If sizeof(long) is 32 bits on 32 bit Linux, then this would not work there as is. Is that an issue?
http://pubs.opengroup.org/onlinepubs/007908799/xsh/time.h.html

Reply:
time_t might be wrong in 32 bit also.

Reply:
If we aren't already working on 32 bit Linux, I suspect there will be a number of structs to modify for it.

Reply:
Looks like check_type_size could help us in that case. I am not sure it is worth doing this without a system to test on.

Reply:
From https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md :

- Don't cheat and take advantage of coincidental agreement between one flavor's ABI and the shim's ABI.
- Sick to data types which are guaranteed not to vary in size across flavors.